File: nodes/null-bomb/null-bomb.process.js
export default async ({inputs, settings, config}) => {

  const value = inputs.value;
  const nullProbability = parseFloat(inputs.nullProbability) || 0.1; // Default to 10% chance
  
  // Validate probability is between 0 and 1
  const probability = Math.max(0, Math.min(1, nullProbability));
  
  // Generate random number between 0 and 1
  const random = Math.random();
  
  // Return null/undefined if random number is less than probability
  if (random < probability) {
    // Randomly choose between null and undefined for variety
    const nullType = Math.random() < 0.5 ? null : undefined;
    return { result: nullType };
  }
  
  // Otherwise, pass through the original value
  return { result: value };
}; 

File: nodes/null-bomb/null-bomb.process.py
import random

async def process(inputs, settings, config, nodeConfig):
    """
    Process function for the Null Bomb node.
    Randomly returns None based on probability, or passes through the original value.
    """
    value = inputs.get("value")
    null_probability = float(inputs.get("nullProbability", 0.1))  # Default to 10% chance
    
    # Validate probability is between 0 and 1
    probability = max(0.0, min(1.0, null_probability))
    
    # Generate random number between 0 and 1
    random_val = random.random()
    
    # Return None if random number is less than probability
    if random_val < probability:
        return {"result": None}
    
    # Otherwise, pass through the original value
    return {"result": value} 